src/lib/yang/schema.lua

```diff
@@ -564,6 +570,24 @@ local primitive_types = set(
    'binary', 'bits', 'boolean', 'decimal64', 'empty', 'enumeration',
    'identityref', 'instance-identifier', 'leafref', 'string', 'union')
 
+-- Inherits config attributes from parents
+local function inherit_config(schema, config)
+   if schema.config ~= nil then
+      assert(not config or schema.config == false)
+      config = schema.config
+   elseif config ~= nil then
+      schema.config = config
+   end
+
+   if schema.body then
+      for name, node in pairs(schema.body) do
```

Comment:
Sorry, I miscommunicated.  Please remove the shallow_copy below and instead at line 583 add "node.body = shallow_copy(node.body).  Thanks :)